src/repository/models.py

```diff
@@ -281,6 +281,12 @@ class Preprint(models.Model):
         null=True,
         on_delete=models.SET_NULL,
     )
+    submission_supplementary_files = models.ManyToManyField(
```

Comment:
I think a one-to-many relationship works best here, even if it leads to some duplication across multiple articles. Some examples:
- removing a supplementary file from a preprint: We need to check all the pointers from the supplementary file or risk having orphan supplementary files.
- Ordering supplementary files: If we want the authors to re-order the supplementary files, we can do so with an attribute on the `SupplementaryPreprintFile` model. With a M2M we would need an intermediate model that holds this attribute.